src/task1/common/imagemetadata.py

```diff
@@ -0,0 +1,116 @@
+# Sample Call
+#
+# p = MetadataProcessor()
+# r = c.ReadForProcessing("test/2018-05-25_16-22-29-018.txt")
+# p.Process(r)
+
+
+# Sample Output
+#
+# {
+#   "corrected": {
+#     "gps": {
+#       "longitude": "-79.471321",
+#       "latitude": "43.750072",
+#       "altitude_agl": "57.169998",
+#       "timestamp": "2018-05-25_16-22-28-918"
+#     },
+#     "altitude": {
+#       "roll_angle": "-9.738050",
+#       "pitch_angle": "-2.027207",
+#       "yaw_angle": "-52.516975",
+#       "timestamp": "2018-05-25_16-22-28-197"
+#     },
+#     "timestamp": "307819"
+#   },
+#   "uncorrected": {
+#     "gps": {
+#       "longitude": "-79.471306",
+#       "latitude": "43.750061",
+#       "altitude_agl": "56.959999",
+#       "timestamp": "2018-05-25_16-22-28-918"
+#     },
+#     "altitude": {
+#       "roll_angle": "-7.079698",
+#       "pitch_angle": "-1.237615",
+#       "yaw_angle": "-51.876190",
+#       "timestamp": "2018-05-25_16-22-28-197"
+#     },
+#     "timestamp": "307599"
+#   },
+#   "index": "0"
+# }
+#
+# The corrected data should be used over the uncorrected data, but is left in for debugging reasons.
+#
+# If the index is 0, the data is extrapolated, else data is inteprolated with number being the index within the queue
+# See https://github.com/UTAT-AeRo/TeledyneDalsa_Odroid/issues/2
+
+
+class MetadataProcessor(object):
+    """Processor for the metadata generated by the image capture. https://github.com/UTAT-AeRo/TeledyneDalsa_Odroid"""
+
+    def Read(self, path):
+        """Returns a string of text from the given file path
+
+        Keyword arguments:
+        path - the relative path to the file
```

Comment:
May make more sense to have this pass a path to the image rather than the metadata file. This would off load the responsibility of finding the metadata file to this class rather than the class using it. It might be useful if we want to store the  metadata separately.